Bug: replication diagnostics conflict and corrupted transaction log after restart in v1.20.0

## Summary

After restarting a ReductStore 1.20.0 instance with active replication/lifecycle configuration, replication state for `orion-to-play` becomes corrupted.

Observed symptoms:

1. Replication diagnostics persistence starts failing with duplicate timestamps:
   - `Failed to persist replication diagnostics for 'orion-to-play': [Conflict] A record with timestamp ... already exists`
2. The replication transaction log is treated as corrupted and repeatedly dropped.
3. ReductStore then fails to recreate the transaction log because the expected `$meta` path does not exist.

This looks like a restart/recovery bug in replication state handling.

## Version

- ReductStore Core `1.20.0`
- Build info from startup log: `ReductStore Core 1.20.0 [ at 2026-06-16T14:43:34Z]`

## Configuration

Relevant instance config from startup log:

- `RS_INSTANCE_NAME = orion`
- `RS_DATA_PATH = /data`
- `RS_REPLICATION_2_NAME = orion-to-play`
- `RS_REPLICATION_2_SRC_BUCKET = ros_data`
- `RS_REPLICATION_2_DST_BUCKET = orion`
- `RS_REPLICATION_2_DST_HOST = https://play.reduct.store`
- `RS_REPLICATION_2_ENTRIES = right*,Pablo*,imu,tf,rosout`
- `RS_REPLICATION_2_WHEN = {"$each_t":"500ms","&gps_z":{"$gt":150}}`
- `RS_LIFECYCLE_1_NAME = ros-data-compress`
- `RS_LIFECYCLE_1_TYPE = compress`
- `RS_LIFECYCLE_1_BUCKET = ros_data`
- `RS_LIFECYCLE_1_OLDER_THAN = 2h`

Buckets include `ros_data` with FIFO quota enabled.

## What happened

After restart, replication to `https://play.reduct.store/orion` initially reports the remote server as unavailable during startup, which is expected.

Later, replication recovery enters a broken state:

- diagnostics writes start failing due to duplicate timestamps
- transaction log reads fail because the log file is missing
- the task declares the log corrupted and tries to recreate it
- recreation fails because the `$meta` directory/path does not exist
- the same error loop repeats continuously

## Key log excerpts

### Startup

```text
2026-06-17 05:50:04.719 [INFO] ReductStore Core 1.20.0 [ at 2026-06-16T14:43:34Z]
```

### Diagnostics conflicts after restart

```text
2026-06-17 04:58:14.995 [ERROR] Failed to persist replication diagnostics for 'orion-to-play': [Conflict] A record with timestamp 1781672289877972 already exists
2026-06-17 05:03:10.434 [ERROR] Failed to persist replication diagnostics for 'orion-to-play': [Conflict] A record with timestamp 1781672585318204 already exists
2026-06-17 05:08:34.667 [ERROR] Failed to persist replication diagnostics for 'orion-to-play': [Conflict] A record with timestamp 1781672909554101 already exists
2026-06-17 05:12:50.455 [ERROR] Failed to persist replication diagnostics for 'orion-to-play': [Conflict] A record with timestamp 1781673169601869 already exists
2026-06-17 05:45:54.080 [ERROR] Failed to persist replication diagnostics for 'orion-to-play': [Conflict] A record with timestamp 1781675148959952 already exists
```

### Recovery loop / corrupted log

```text
2026-06-16 21:23:37.564 [ERROR] Failed to read transaction: ReductError { status: InternalServerError, message: "File \"/data/ros_data/Pablo05/odom/$meta/orion-to-play.log\" does not exist" }
2026-06-16 21:23:37.564 [INFO] Transaction log is corrupted, dropping the whole log
2026-06-16 21:23:37.566 [ERROR] Failed to remove transaction log: ReductError { status: InternalServerError, message: "No such file or directory (os error 2)" }
2026-06-16 21:23:37.566 [INFO] Creating a new transaction log: "/data/ros_data/Pablo05/odom/$meta/orion-to-play.log"
2026-06-16 21:23:37.566 [ERROR] Failed to create transaction log: ReductError { status: InternalServerError, message: "No such file or directory (os error 2)" }
```

The same sequence repeats every few seconds afterward.

### Remote bucket startup errors seen before corruption

```text
2026-06-16 21:09:51.091 [ERROR] Failed to get remote bucket https://play.reduct.store/orion: [ServiceUnavailable] The server is starting up, please try again later
```

## Expected behavior

After restart, replication should either:

- recover its diagnostics and transaction logs cleanly, or
- recreate missing state (including metadata path) without entering a permanent error loop.

It should not continuously hit duplicate diagnostic timestamps or fail to recreate its own transaction log.

## Actual behavior

Replication for `orion-to-play` gets stuck in a persistent broken state after restart, with repeated conflict errors and repeated failure to recreate the missing transaction log.

## Notes

Possible areas to inspect:

- diagnostic timestamp generation/idempotency during restart/recovery
- replication transaction log recovery when `$meta` path is absent
- ordering/race conditions between lifecycle/compression and replication metadata/log handling
- behavior when remote bucket is temporarily unavailable during startup

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -7,6 +7,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Fixed
+
+- Fix replication diagnostics timestamp conflicts and recover corrupted transaction logs created on demand for `$meta` notifications, [PR-1452](https://github.com/reductstore/reductstore/pull/1452)
+
 ## 1.20.0 - 2026-06-16
 
 ### Breaking Changes

--- a/reductstore/src/replication/replication_aggregator.rs
+++ b/reductstore/src/replication/replication_aggregator.rs
@@ -84,6 +84,7 @@ pub(super) struct ReplicationEventAggregator {
     sink: SystemEventSink,
     replication_name: String,
     active: Option<ReplicationAggregate>,
+    last_event_timestamp: u64,
 }
 
 impl ReplicationEventAggregator {
@@ -92,7 +93,14 @@ impl ReplicationEventAggregator {
             sink,
             replication_name,
             active: None,
+            last_event_timestamp: 0,
         }
+    }
+
+    fn next_event_timestamp(&mut self) -> u64 {
+        let timestamp = now_micros().max(self.last_event_timestamp + 1);
+        self.last_event_timestamp = timestamp;
+        timestamp
     }
 
     /// Record one replication pass as a list of `(result, records, data_size)`.
@@ -125,7 +133,6 @@ impl ReplicationEventAggregator {
             return;
         }
 
-        let timestamp = now_micros();
         let mut first = true;
         for (status, (records, data_size, message)) in per_status {
             let pass_duration = if first { duration } else { 0.0 };
@@ -157,6 +164,7 @@ impl ReplicationEventAggregator {
                     aggregate.flush_at = now + Duration::from_secs(AGGREGATION_WINDOW_SECS);
                 }
                 None => {
+                    let timestamp = self.next_event_timestamp();
                     let success = is_success(status);
                     self.active = Some(ReplicationAggregate {
                         status,
@@ -306,6 +314,26 @@ mod tests {
         assert_eq!(captured[1].payload["failed_records"], 4);
         assert_eq!(captured[1].payload["written_records"], 0);
         assert_eq!(captured[1].message, "missing");
+    }
+
+    #[rstest]
+    #[tokio::test]
+    async fn mixed_status_pass_uses_unique_event_timestamps(events: Arc<Mutex<Vec<SystemEvent>>>) {
+        let mut agg = aggregator(Arc::clone(&events), false);
+
+        agg.record_pass(
+            1,
+            0.1,
+            &[(Ok(()), 1, 10), (Err(not_found!("missing")), 1, 0)],
+        )
+        .await;
+        agg.record_pass(0, 0.1, &[(Ok(()), 1, 10)]).await;
+
+        let captured = events.lock().unwrap();
+        assert_eq!(captured.len(), 2);
+        assert_eq!(captured[0].status, 200);
+        assert_eq!(captured[1].status, 404);
+        assert_ne!(captured[0].timestamp, captured[1].timestamp);
     }
 
     /// Schema invariant (#4): success and failure events carry the identical

--- a/reductstore/src/replication/replication_task.rs
+++ b/reductstore/src/replication/replication_task.rs
@@ -186,27 +186,12 @@ impl ReplicationTask {
                         &entry.name,
                         &replication_name,
                     );
-                    let log = match TransactionLog::try_load_or_create(
+                    let log = Self::load_or_recreate_transaction_log(
                         &path,
                         thr_system_options.transaction_log_size,
+                        &entry.name,
                     )
-                    .await
-                    {
-                        Ok(log) => log,
-                        Err(err) => {
-                            error!(
-                                "Failed to load transaction log for entry '{}': {:?}",
-                                entry.name, err
-                            );
-                            info!("Creating a new transaction log for entry '{}'", entry.name);
-                            FILE_CACHE.remove(&path).await?;
-                            TransactionLog::try_load_or_create(
-                                &path,
-                                thr_system_options.transaction_log_size,
-                            )
-                            .await?
-                        }
-                    };
+                    .await?;
 
                     logs.insert(entry.name, Arc::new(AsyncRwLock::new(log)));
                 }
@@ -398,14 +383,16 @@ impl ReplicationTask {
         // because it is used by the replication thread
         let exists = { self.log_map.read().await?.contains_key(&entry_name) };
         if !exists {
-            let log = TransactionLog::try_load_or_create(
-                &Self::build_path_to_transaction_log(
-                    self.storage.data_path(),
-                    &self.settings.src_bucket,
-                    &entry_name,
-                    &self.name,
-                ),
+            let path = Self::build_path_to_transaction_log(
+                self.storage.data_path(),
+                &self.settings.src_bucket,
+                &entry_name,
+                &self.name,
+            );
+            let log = Self::load_or_recreate_transaction_log(
+                &path,
                 self.system_options.transaction_log_size,
+                &entry_name,
             )
             .await?;
             let mut map = self.log_map.write().await?;
@@ -515,6 +502,25 @@ impl ReplicationTask {
         name: &str,
     ) -> PathBuf {
         storage_path.join(format!("{}/{}/{}.log", bucket, entry, name))
+    }
+
+    async fn load_or_recreate_transaction_log(
+        path: &PathBuf,
+        transaction_log_size: usize,
+        entry_name: &str,
+    ) -> Result<TransactionLog, ReductError> {
+        match TransactionLog::try_load_or_create(path, transaction_log_size).await {
+            Ok(log) => Ok(log),
+            Err(err) => {
+                error!(
+                    "Failed to load transaction log for entry '{}': {:?}",
+                    entry_name, err
+                );
+                info!("Creating a new transaction log for entry '{}'", entry_name);
+                FILE_CACHE.remove(path).await?;
+                TransactionLog::try_load_or_create(path, transaction_log_size).await
+            }
+        }
     }
 
     fn load_mode(&self) -> ReplicationMode {
@@ -699,6 +705,40 @@ mod tests {
                 is_provisioned: false,
                 pending_records: 0,
             }
+        );
+    }
+
+    #[rstest]
+    #[tokio::test]
+    async fn test_add_meta_entry_with_broken_transaction_log(
+        mut remote_bucket: MockRmBucket,
+        mut notification: TransactionNotification,
+        settings: ReplicationSettings,
+        path: PathBuf,
+    ) {
+        remote_bucket.expect_is_active().return_const(false);
+        let mut replication = build_replication(path, remote_bucket, settings.clone()).await;
+
+        notification.entry = "test1/$meta".to_string();
+        let log_path = ReplicationTask::build_path_to_transaction_log(
+            replication.storage.data_path(),
+            &settings.src_bucket,
+            &notification.entry,
+            replication.name(),
+        );
+
+        fs::create_dir_all(log_path.parent().unwrap()).unwrap();
+        let mut log_file = fs::File::create(&log_path).unwrap();
+        log_file.write_all(&916usize.to_be_bytes()).unwrap();
+        log_file.write_all(&16usize.to_be_bytes()).unwrap();
+        log_file.set_len(916).unwrap();
+
+        replication.notify(notification).await.unwrap();
+
+        assert_eq!(fs::metadata(&log_path).unwrap().len(), 9016);
+        assert_eq!(
+            get_entries_from_transaction_log(&mut replication, "test1/$meta").await,
+            vec![Transaction::WriteRecord(10)]
         );
     }
 

